src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -307,8 +340,9 @@ public void paint(final Graphics g, final JComponent c) {
         // performs icon and text rect calculations
         final String text;
         final View v = (View)c.getClientProperty(BasicHTML.propertyKey);
-        if (v != null) {
-            // use zero insets for view since layout only handles text calculations
+        if (v != null && b.getIcon() == null) {
```

Comment:
I think you can store `b.getIcon` same as `text` variable and reuse it down below too as it is now being called more than once

Reply:
Good idea. Added